knowledge-base/create-tables-programmatically.md

```diff
@@ -0,0 +1,161 @@
+detailSection.Name = "detail";
+
+var pageFooterSection = new Telerik.Reporting.PageFooterSection();
+pageFooterSection.Height = Telerik.Reporting.Drawing.Unit.Cm(2.5D);
+pageFooterSection.Name = "pageFooterSection";
+
+report.Items.AddRange(new Telerik.Reporting.ReportItemBase[] { pageHeaderSection, detailSection, pageFooterSection });
+````
+
+### 3. Preparing the Data Source
+
+Telerik Reporting supports various data sources including JSON, CSV, SQL, business objects, and others. For more information, check the article [Data Source Components at a Glance]({%slug telerikreporting/designing-reports/connecting-to-data/data-source-components/overview%}) This example uses an [SqlDataSource component]({%slug telerikreporting/designing-reports/connecting-to-data/data-source-components/sqldatasource-component/overview%}):
+
+````csharp
+string selectCommand = "SELECT * FROM production.productphoto;";
+string connectionString = "server=localhost\\sqlexpress;database=AdventureWorks2022;trusted_connection=true;";
+var sqlDataSource = new Telerik.Reporting.SqlDataSource(connectionString, selectCommand);
+sqlDataSource.Name = "sqlDataSource";
+sqlDataSource.ProviderName = "System.Data.SqlClient";
+````
+
+## Solution
+
+Once the report structure and data source are ready, you can proceed with creating the table.
+
+### 1. Create the Table
+
+Instantiate a `Telerik.Reporting.Table` object, assign the pre-created data source, and add the table to the detail section:
+
+````csharp
+var table = new Telerik.Reporting.Table();
+table.Name = "table";
+table.Location = new Telerik.Reporting.Drawing.PointU(Telerik.Reporting.Drawing.Unit.Cm(1.5D), Telerik.Reporting.Drawing.Unit.Cm(1.5D));
+table.DataSource = sqlDataSource;
+detailSection.Items.Add(table);
+````
+
+### 2. Add a Detail Row Group
+
+Add a detail row group and a body row.
+
+````csharp
+var rowGroup = new Telerik.Reporting.TableGroup();
+rowGroup.Name = "detailTableGroup";
+rowGroup.Groupings.Add(new Telerik.Reporting.Grouping(null));
+table.RowGroups.Add(rowGroup);
+table.Body.Rows.Add(new Telerik.Reporting.TableBodyRow(Telerik.Reporting.Drawing.Unit.Cm(0.609D)));
+````
+
+>note Adding a detail row group is mandatory for the table to display data. When using a report designer, this group is created automatically. To mark the group as a detail group programmatically, use a null grouping as shown above.
+
+### 3. Add Columns Dynamically Based on Data
+
+The method for obtaining field names for your table columns depends on the type of data source you are using. In this example, since an `SqlDataSource` is utilized, the code demonstrates how to extract column names directly from an SQL database. If your data source is a CSV, JSON, or another format, you will need to adjust the logic to extract field names appropriately and create the string collection named `columnNames`. Once you have the field names, the process for dynamically generating table columns remains consistent with the below code snippet.
```

Comment:
```suggestion
The method for obtaining field names for your table columns depends on the type of data source you are using. In this example, since an `SqlDataSource` is utilized, the code below demonstrates how to extract column names directly from an SQL database. 

> If your data source is a CSV, JSON, or another format, adjust the logic to extract field names and create the string collection named `columnNames`. Once you have the field names, the process for dynamically generating table columns remains consistent with the code snippet below.
```